src/components/Dashboard /StudentData/index.js

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useState } from "react";
+import axios from "axios";
+import { useSelector } from "react-redux";
+// import { METHODS } from "../../services/api";
+import moment from "moment";
+import { Link } from "react-router-dom";
+import "./styles.scss";
+
+const getPartnerIdFromUrl = () => {
+  let partnerId;
+  if (window.location.pathname.includes("partners")) {
+    partnerId = window.location.pathname.split("/").pop();
+  }
+  return partnerId;
+};
+
+function StudentData() {
+  const [students, setStudents] = useState([]);
+  const [searchTerm, setSearchTerm] = useState("");
+  const user = useSelector(({ User }) => User);
+
+  useEffect(() => {
+    let id = getPartnerIdFromUrl();
+    axios
+      .get(`https://api.merakilearn.org/partners/${id}/users`, {
+        headers: { Authorization: user.data.token },
+      })
+      .then((res) => {
+        setStudents(res.data);
+      });
+  }, []);
+
+  const handleSearchChange = (e) => {
+    e.preventDefault();
+    setSearchTerm(e.target.value);
+  };
+
+  return (
+    <>
+      <div className="table-search">
+        <input
+          type="text"
+          placeholder="Search..."
+          onChange={handleSearchChange}
+          value={searchTerm}
+        />
+      </div>
+      <table className="student-overview-table">
+        <thead>
+          <tr>
+            <th>Students Name</th>
+            <th>Enrolled On </th>
+            <th>Number of class attend</th>
+            <th>Last class title</th>
+            <th>Last class attended </th>
+          </tr>
+        </thead>
+
+        {students.length > 0 &&
```

Comment:
there should be an empty state with message 'There are no results to display' in case there is no data to show in the table.

Reply:
Done